core/src/main/java/com/graphhopper/storage/GraphHopperStorage.java

```diff
@@ -420,4 +452,14 @@ public Graph copyTo(Graph g) {
     public GraphExtension getExtension() {
         return baseGraph.getExtension();
     }
+
+    private Collection<CHGraphImpl> getAllCHGraphs() {
+        // todo: this method is only used to have a 'view' on the two collections. we could also create this only once
+        // as long as the graph collections are only modified in the constructor (otherwise we would have to make sure
+        // the lists are in sync). another option would be something like guava concat.
+        List<CHGraphImpl> result = new ArrayList<>();
+        result.addAll(nodeBasedCHGraphs);
+        result.addAll(edgeBasedCHGraphs);
```

Comment:
A minor optimization here is to create the new ArrayList with the nodeBasedCHGraphs already or at least init the size with node+edge

Reply:
I did not want to create the list already, because then one has to make sure that it is in sync with the other two. Right now this is easy because the lists are changed only in the constructor, but I was afraid might change at some point and thought it is not worth it. I do not think this method is never called in a performance-relevant code. I added the capacity parameter to the ArrayList constructor for now.